lib/createExplorer.js

```diff
@@ -74,11 +71,9 @@ module.exports = function (options) {
       })
       .then((result) => {
         if (result) return result;
-
-        const nextDirectory = joinPath(splitPath(directory).slice(0, -1));
-        if (!nextDirectory || directory === options.stopDir) return null;
-
-        return searchDirectory(nextDirectory);
+        if (!splitSearchPath) splitSearchPath = splitPath(directory);
+        if (!splitSearchPath.pop() || directory === options.stopDir) return null;
+        return searchDirectory(joinPath(splitSearchPath), splitSearchPath);
```

Comment:
In Windows `path.resolve("E:")` return a string path that same with `process.cwd()`.
And `process.cwd().split(path.sep)` return `[ 'E:', 'work', 'cosmiconfig' ]`

So in last step of loop, `path.resolve("E:")` will run. It make a infinite loops.

There is a demo
```javascript
function walk(dir) {
	dir = path.resolve(dir).split(path.sep);
	console.log(dir);
	walk(dir.slice(0, -1).join(path.sep));
}

walk(process.cwd());
```
It will run indefinitely on Windows.